src/list/list.test.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import {render, screen} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+
+import {testA11y} from "../core/utils/test/testUtils";
+import List, {ListProps} from "./List";
+import ListItem from "./item/ListItem";
+
+describe("<List />", () => {
+  const listItems = ["Hipo", "Labs"];
+  const defaultListProps: ListProps = {
+    testid: "list",
+    items: listItems,
+    children: (item, testId, index) => (
+      <ListItem key={`list-item-${index}`} testid={testId}>
+        <span>{item}</span>
+      </ListItem>
+    )
+  };
+
+  it("should render correctly", () => {
+    render(<List {...defaultListProps} />);
+  });
+
+  it("should match snapshot", () => {
+    const tree = create(<List {...defaultListProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<List {...defaultListProps} />);
+
+    await testA11y(container);
+  });
+
+  it("should render children correctly", () => {
+    const {getByTestId} = render(<List {...defaultListProps} />);
+
+    expect(getByTestId(defaultListProps.testid!).children).toHaveLength(listItems.length);
+
+    for (let index = 0; index < listItems.length; index++) {
+      const listItemContent = getByTestId(`list.item-${index}`);
+
+      expect(getByTestId(defaultListProps.testid!).children[index]).toEqual(
+        listItemContent
+      );
+    }
+  });
+
+  it("should render placeholder correctly", () => {
+    const {getByTestId} = render(
+      <List
+        placeholderProps={{
+          shouldDisplayPlaceholder: true,
+          placeholder: <p data-testid={"list.placeholder"}>{"Loading..."}</p>
+        }}
+        {...defaultListProps}
+      />
+    );
+
+    expect(getByTestId(defaultListProps.testid!)).toContainElement(

```

Comment:
Cant you use the container that is returned by the render function here ? Same goes for the other test blocks
```suggestion
    expect(container).toContainElement(

```